server/src/classes/Player.cpp

```diff
@@ -495,6 +495,158 @@ static void SetWeaponAmmo(js::FunctionContext& ctx)
     player->SetWeaponAmmo(weaponHash, ammoCount);
 }
 
+static void GetAmmoSpecialType(js::FunctionContext& ctx)
+{
+    if(!ctx.CheckThis()) return;
+    if(!ctx.CheckArgCount(1)) return;
+    alt::IPlayer* player = ctx.GetThisObject<alt::IPlayer>();
+
+    uint32_t ammoHash;
+    if(!ctx.GetArgAsHash(0, ammoHash)) return;
+
+    ctx.Return(player->GetAmmoSpecialType(ammoHash));
+}
+
+static void SetAmmoSpecialType(js::FunctionContext& ctx)
+{
+    if(!ctx.CheckThis()) return;
+    if(!ctx.CheckArgCount(2)) return;
+    alt::IPlayer* player = ctx.GetThisObject<alt::IPlayer>();
+
+    uint32_t ammoHash;
+    if(!ctx.GetArgAsHash(0, ammoHash)) return;
+
+    alt::AmmoSpecialType ammoSpecialType;
+    if(!ctx.GetArg(1, ammoSpecialType)) return;
+
+    player->SetAmmoSpecialType(ammoHash, ammoSpecialType);
+}
+
+static void GetAmmoFlags(js::FunctionContext& ctx)
+{
+    if(!ctx.CheckThis()) return;
+    if(!ctx.CheckArgCount(1)) return;
+    alt::IPlayer* player = ctx.GetThisObject<alt::IPlayer>();
+
+    uint32_t weaponHash;
+    if(!ctx.GetArgAsHash(0, weaponHash)) return;
+
+    alt::AmmoFlags flags = player->GetAmmoFlags(weaponHash);
+
+    js::Object obj;
+    obj.Set("infiniteAmmo", flags.infiniteAmmo);
+    obj.Set("addSmokeOnExplosion", flags.addSmokeOnExplosion);
+    obj.Set("fuse", flags.fuse);
+    obj.Set("fixedAfterExplosion", flags.fixedAfterExplosion);
+
+    ctx.Return(obj);
+}
+
+static void SetAmmoFlags(js::FunctionContext& ctx)
+{
+    if(!ctx.CheckThis()) return;
+    if(!ctx.CheckArgCount(5)) return;
+    alt::IPlayer* player = ctx.GetThisObject<alt::IPlayer>();
+
+    uint32_t weaponHash;
+    if(!ctx.GetArgAsHash(0, weaponHash)) return;
+
+    bool infiniteAmmo;
+    if(!ctx.GetArg(1, infiniteAmmo)) return;
+
+    bool addSmokeOnExplosion;
+    if(!ctx.GetArg(2, addSmokeOnExplosion)) return;
+
+    bool fuse;
+    if(!ctx.GetArg(3, fuse)) return;
+
+    bool fixedAfterExplosion;
+    if(!ctx.GetArg(4, fixedAfterExplosion)) return;
+
+    player->SetAmmoFlags(weaponHash, { infiniteAmmo, addSmokeOnExplosion, fuse, fixedAfterExplosion });
```

Comment:
Please use the more clearer syntax
```suggestion
    player->SetAmmoFlags(weaponHash, alt::AmmoFlags{ infiniteAmmo, addSmokeOnExplosion, fuse, fixedAfterExplosion });
```